lib/rubocop/cop/commissioner.rb

```diff
@@ -7,6 +7,35 @@ module Cop
     class Commissioner
       include RuboCop::AST::Traversal
 
+      # How a Commissioner returns the results of the investigation
+      # as a list of Cop::InvestigationReport and any errors caught
+      # during the investigation.
+      # Immutable
+      # Consider creation API private
```

Comment:
TBH, the only reason why I wrote this is I'd like to `keyword_argument: true` for our Structs, but that's Ruby 2.5+ only. I'd be really tempted to use my `backport` for this...

Nothing is truly private in Ruby, and for `Cop::Base` in particular since we expect people to write Cops. I tried moving some of the code into `Investigation` but gave up for now.